lib/skate/detours/detours.ex

```diff
@@ -193,36 +193,33 @@ defmodule Skate.Detours.Detours do
   end
 
   @doc """
-  Update or create a detour given a user id & detour id.
+  Update or insert a detour given a user id and a XState Snapshot.
   """
-  def update_or_create_detour_for_user(user_id, uuid, attrs \\ %{}) do
-    user = User.get_by_id!(user_id)
-
-    previous_record = uuid != nil && Skate.Repo.get(Detour, uuid)
-
-    detour =
-      case uuid do
-        nil ->
-          create_detour_for_user(user_id, attrs)
```

Comment:
Huh! I'm realizing that this case wasn't necessary, given the Repo.insert would handle that, right?

Reply:
Affirmative! I tried to figure it[^1] out when we first added this but ran into issues, but it went smoothly this time so 🤷🏻‍♀️ 

[^1]: "it" being `upsert` with a single call without a `case`